ui/helpers/utils/util.js

```diff
@@ -443,3 +443,77 @@ export const toHumanReadableTime = (t, milliseconds) => {
 export function clearClipboard() {
   window.navigator.clipboard.writeText('');
 }
+
+const solidityTypes = () => {
+  const types = [
+    'bool',
+    'address',
+    'string',
+    'bytes',
+    'int',
+    'uint',
+    'fixed',
+    'ufixed',
+  ];
+
+  const ints = Array.from(new Array(32)).map(
+    (_, index) => `int${(index + 1) * 8}`,
+  );
+  const uints = Array.from(new Array(32)).map(
+    (_, index) => `uint${(index + 1) * 8}`,
+  );
+  const bytes = Array.from(new Array(32)).map(
+    (_, index) => `bytes${index + 1}`,
+  );
+  return [...types, ...ints, ...uints, ...bytes];
+};
+
+export const sanitizeMessage = (msg, baseType, types) => {
+  let baseTypeDefinitions;
+  if (types) {
+    baseTypeDefinitions = types[baseType];
+    if (!baseTypeDefinitions) {
+      return msg;
+    }
+  } else {
+    return msg;
+  }
+
+  const sanitizedMessage = {};
+  const msgKeys = Object.keys(msg);
+  msgKeys.forEach((msgKey) => {
+    const definedType = Object.values(baseTypeDefinitions).find(
+      (baseTypeDefinition) => baseTypeDefinition.name === msgKey,
+    );
+
+    if (definedType) {
+      // key has a type. check if the definedType is also a type
+      const nestedTypeDefinition = types[definedType.type.replace('[]', '')];
+      if (nestedTypeDefinition) {
+        if (definedType.type.indexOf('[]') > 0) {
+          // nested array
+          const definedArrayType = definedType.type.replace('[]', '');
+          sanitizedMessage[msgKey] = msg[msgKey].map((value) =>
+            sanitizeMessage(value, definedArrayType, types),
+          );
+        } else {
+          // nested object
+          sanitizedMessage[msgKey] = sanitizeMessage(
+            msg[msgKey],
+            definedType.type,
+            types,
+          );
+        }
+      } else {
+        // check if it's a valid solidity type
+        const isSolidityType = solidityTypes().includes(
+          definedType.type.replace('[]', ''),
```

Comment:
We should exclude types with `[]` at the beginning or middle as well:
```suggestion
          definedType.type.replace(\[\]$/, ''),
```